pkg/kv/cosmosdb/store.go

```diff
@@ -342,8 +348,9 @@ func (s *Store) Scan(ctx context.Context, partitionKey []byte, options kv.ScanOp
 		encoding:     encoding,
 	}
 	it.runQuery()
-	if it.err != nil {
-		return nil, it.err
+	err := it.Err()
+	if err != nil {
+		return nil, err
```

Comment:
not related to this change but - having runQuery change the state of the iterator error and not communicate it, can lead to bugs.
like next where we return false in case we fail or complete and the caller need to check err.